### DISCUSSION TITLE
participant-initiated ITI

### DISCUSSION THREAD

**Discussion by hariPurohit** [jspsych/jsPsych#1189] [Q&A] [answered]
Hi!

I am coding one of my project experiments using jsPsych. My experiment design is a simple RDK task, based on Experiment 3 from Ratcliff, Voskuilen and McKoon (2018) paper. In my design, I wish to have a self-initiated ITI i.e. instead of the next trial starting after 'x' milliseconds, i would like the trial to begin only when the participant is ready, via a buttton-press. 

I am aware that there's default_iti and post-trial gap parameters to allow me to set the ITI time but is it possible instead to only begin the nth trial based on a button-press, after the previous (n-1) trial has ended?

**Comment by jodeleeuw** [comment, marked as answer]
Hi @hariPurohit ,

I recommend just putting a "trial" in between. If you want to begin the next trial on a keyboard press you could stick this trial in between:

```js
var wait = {
	type: 'html-keyboard-response',
	stimulus: '<p>Press any key to start the next trial.</p>'
}
```

You could also show this trial with a blank stimulus, if you just want a blank screen while waiting for the key press.

**Reply by hariPurohit** [reply]
Yes, that makes sense..thanks!
